tests/phpunit/applaunch_test.php

```diff
@@ -0,0 +1,225 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Test the applaunch persistent class.
+ *
+ * @package    mod_applaunch
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2021 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+use mod_applaunch\app_type;
+use mod_applaunch\applaunch;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once($CFG->dirroot . '/mod/applaunch/lib.php');
```

Comment:
Any reason this isn't autoloaded? not sure if this is a mod specific thing...

Reply:
It's not a class, as it's the main lib file. I don't usually touch them so wasn't sure if they were autoloaded or not?

Reply:
lib.php specifically has some magic afaik, along with version.php. Try it and see?

Reply:
You're right. Good to know. Removed require.